tests/integration/testsuites/upgrade/scenario.go

```diff
@@ -146,12 +155,16 @@ func (s *scenario) upgradeApiGateway() error {
 		return err
 	}
 
-	var dep v1.Deployment
+	var dep appsv1.Deployment
 	err = runtime.DefaultUnstructuredConverter.FromUnstructured(apiGatewayDeployment.UnstructuredContent(), &dep)
 	if err != nil {
 		return err
 	}
 
+	if dep.Spec.Template.Spec.Containers[0].Image == s.APIGatewayImageVersion {
+		return errors.New("trying to upgrade to same version of api-gateway controller")
```

Comment:
I will prefer to fail, since if we run the test with wrong image version (same as from kyma deploy) and expected upgrade which did not happen.

it may be logged but no one will see it and consider test did upgrade, but it actually did not. When we have a scenario when we want to test upgrade to same version this could be adjusted or done in different func. But this test it is essential that we actually doing an upgrade and this will red flag it if we did not configure and run it properly.

Reply:
OK, let's leave it, this shouldn't really be a blocker as we are building new image every PR either way
Only might happen if we do something with release tagged images, but it's an edge case